src/swap/SwapReviewScreen.tsx

```diff
@@ -0,0 +1,364 @@
+import { SwapEvents } from 'src/analytics/Events'
+import ValoraAnalytics from 'src/analytics/ValoraAnalytics'
+import { ErrorMessages } from 'src/app/ErrorMessages'
+import {
+  maxSwapSlippagePercentageSelector,
+  swapFeeEnabledSelector,
+  swapFeePercentageSelector,
+} from 'src/app/selectors'
+import BackButton from 'src/components/BackButton'
+import Button, { BtnSizes } from 'src/components/Button'
+import Dialog from 'src/components/Dialog'
+import CustomHeader from 'src/components/header/CustomHeader'
+import TokenDisplay, { formatValueToDisplay } from 'src/components/TokenDisplay'
+import Touchable from 'src/components/Touchable'
+import InfoIcon from 'src/icons/InfoIcon'
+import { noHeader } from 'src/navigator/Headers'
+import { Screens } from 'src/navigator/Screens'
+import { StackParamList } from 'src/navigator/types'
+import DisconnectBanner from 'src/shared/DisconnectBanner'
+import colors from 'src/styles/colors'
+import fontStyles from 'src/styles/fonts'
+import { Spacing } from 'src/styles/styles'
+import variables from 'src/styles/variables'
+import { Field } from 'src/swap/useSwapQuote'
+import { coreTokensSelector } from 'src/tokens/selectors'
+import { divideByWei, multiplyByWei } from 'src/utils/formatting'
+import Logger from 'src/utils/Logger'
+import networkConfig from 'src/web3/networkConfig'
+import { walletAddressSelector } from 'src/web3/selectors'
+
+type Props = StackScreenProps<StackParamList, Screens.SwapReviewScreen>
+
+const TAG = 'SWAP_REVIEW_SCREEN'
+
+export function SwapReviewScreen(props: Props) {
+  const { toToken, fromToken, swapAmount, updatedField } = props.route.params
+  const [shouldFetch, setShouldFetch] = useState(false)
+  const [estimatedModalVisible, setEstimatedDialogVisible] = useState(false)
+  const [swapFeeModalVisible, setSwapFeeModalVisible] = useState(false)
+  const [swapInfo, setSwapInfo] = useState(null as any)
+  const [fetchError, setFetchError] = useState(false)
+  const coreTokens = useSelector(coreTokensSelector)
+  const walletAddress = useSelector(walletAddressSelector)
+
+  // Items set for remote config
+  const maxSlippagePercent = useSelector(maxSwapSlippagePercentageSelector)
+  const swapFeeEnabled = useSelector(swapFeeEnabledSelector)
+  const swapFeePercentage = useSelector(swapFeePercentageSelector)
+
+  // Remote configs converted to decimals strings
+  // const maxSlippageDecimal = `${maxSlippagePercent / 100}`
+  const swapFeeDecimal = `${swapFeePercentage / 100}`
+
+  const { t } = useTranslation()
+  const dispatch = useDispatch()
+
+  // Token Symbols
+  const toTokenSymbol = coreTokens.find((token) => token.address === toToken)?.symbol
+  const fromTokenSymbol = coreTokens.find((token) => token.address === fromToken)?.symbol
+
+  useEffect(() => {
+    setShouldFetch(true)
+    ValoraAnalytics.track(SwapEvents.swap_review_screen_open, {
+      toToken,
+      fromToken,
+      buyAmount: swapAmount[Field.TO] ?? '',
+    })
+  }, [])
+
+  useAsync(
```

Comment:
you already throw an error inside the promise of `useAwait` when the response returns unsuccessfully, so i think that you can also reply on the `error` return variable from `useAsync` to handle errors instead of a duplicate `fetchError` state variable.

Reply:
I tried this method out and it ended up causing a flicker on the Complete Swap button as seen below in the video.

https://user-images.githubusercontent.com/26950305/185216326-c71055cc-3957-4b7f-b438-8e0b683f76b6.mov

Reply:
ah, you're right. i forgot that the error is cleared when the hook is used again by switching `shouldFetch` to false. how annoying :( let's leave it like this then.

as i'm typing this i'm realising that i encountered this exact problem when implementing the previous screen (like how to persist loading and error states across fetches) - and i remember that's why i built the `useSwapQuote` hook. 🥲 sorry it's just clicked for me. i need to modify the hook a bit so that it can be reused here (mostly how to deal with the `requestUrlRef` which i put in to deal with an edge case specific to the swap amount scree), but i think we can eventually do something like `const { swapInfo, refreshQuote, fetchSwapQuoteError, fetchingSwapQuote } = useSwapQuote()` in this component. this would mean you don't need the useAsync hook here at all and can reuse the logic for fetching from 0x. 

but let's tackle this later, this is good for now!